operator/Makefile

```diff
@@ -174,37 +169,78 @@ everything: build bundle ## Build everything (local binary, operator image, bund
 	$(MAKE) docker-push-index
 
 ##@ Dependencies download
-include $(PROJECT_DIR)/../make/gotools.mk
 
-$(call go-tool, CONTROLLER_GEN, sigs.k8s.io/controller-tools/cmd/controller-gen, tools)
```

Comment:
Sorry, but I consider going from `go-tool` to a simple `go install` to be a regression. The `go-tool` way is a major improvement over how this is done in the upstream template that lacks proper versioning of binaries and ability to have tool versions bumped by dependabot.

Reply:
I was unable to upgrade to 1.33 using the tools `go.mod` after a couple hours of fiddling with the go.mod. There was an unresolvable(?) dependency conflict (on `gnostic` vs `gnostic-models`) that made the whole tools package either uncompilable, or even unable to be go `mod tidy`ed. I think the conflict exists between `kustomize`, `operator-sdk` and `controller-tools`. I was unable to resolve it with rewrites, nor with using various versions of the dependencies. Perhaps that's why `operator-sdk init` sets up new projects this way? 

I understand we cannot have dependencies bumped via dependabot with this method, but in this case, we cannot even update the dependencies manually. Another option would be to have one single `go.mod` for each tool we use. Though, operator-sdk prescribes which versions of the various tools should be used with which operator-sdk versions. I'm not sure how this would behave if we would let dependabot manage those versions? eg [upgrade docs](https://sdk.operatorframework.io/docs/upgrading-sdk-version/v1.25.0/) mentions

> In the Makefile file, replace CONTROLLER_TOOLS_VERSION ?= v0.9.2 with CONTROLLER_TOOLS_VERSION ?= v0.10.0

I'm not sure what's the difference between `go-tool` and `go install`. From what I understand, `go-tool` adds a Makefile template that, in the end, always calls `go install` with the same arguments. Am I missing something?

Reply:
Having a separate `go.mod` per tool might be the way to go, it already seemed necessary for bumping kuttl and I did that there. In fact there are tools in the go ecosystem which help doing just that, but I'm not sure we want to do this just yet.

I remember [having exactly the same reservations](https://github.com/stackrox/stackrox/pull/4649#discussion_r1102767185) about automatically bumping tool versions outside of `operator-sdk` bumps back when this was done, but it turned out to just work great in practice. It makes security-related bumps much more lightweight and easier.